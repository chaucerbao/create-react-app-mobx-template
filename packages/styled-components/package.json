{
  "name": "styled-components-styles",
  "version": "0.1.0",
  "description": "Using styled-components to manage the styles",
  "author": "Alvin Chan <no@email.com>",
  "license": "ISC",
  "stylelint": {
    "processors": [
      "stylelint-processor-styled-components"
    ],
    "extends": [
      "stylelint-config-standard",
      "stylelint-config-concentric-order",
      "stylelint-config-styled-components"
    ]
  },
  "devDependencies": {
    "source-map-explorer": "^1.5.0",
    "stylelint": "^9.2.0",
    "stylelint-config-concentric-order": "^3.1.0",
    "stylelint-config-standard": "^18.2.0",
    "stylelint-config-styled-components": "^0.1.1",
    "stylelint-processor-styled-components": "^1.3.1",
    "tslint": "^5.10.0",
    "webpack-cli": "^2.1.3",
    "webpack-dev-server": "^3.1.4"
  },
  "dependencies": {
    "styled-components": "^3.2.6"
  },
  "scripts": {
    "start": "webpack-dev-server --config ../../webpack.config.js -d",
    "build": "webpack-cli --config ../../webpack.config.js -p --devtool source-map --progress",
    "analyze": "source-map-explorer public/js/script.js{,.map}"
  }
}
