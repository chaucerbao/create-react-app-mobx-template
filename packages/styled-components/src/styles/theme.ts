// tslint:disable object-literal-sort-keys

// Type definitions
export interface ITheme {
  breakpoints: IDictionary
  colors: IDictionary
  spacers: IDictionary
}
interface IDictionary {
  [key: string]: number | string
}

// Theme
const breakpoints = {
  mobile: '0',
  tablet: '768px',
  desktop: '1280px',
}

const colors = {
  primary: '#444',
  grey: '#888',
  white: '#fafafa',
}

const spacers = {
  xs: '4px',
  sm: '8px',
  md: '16px',
  lg: '24px',
  xl: '32px',
}

const sort = (dictionary: IDictionary) =>
  Object.entries(dictionary)
    .sort((a, b) => {
      const aValue = parseFloat(a[1].toString())
      const bValue = parseFloat(b[1].toString())

      if (aValue < bValue) {
        return -1
      }
      if (aValue > bValue) {
        return 1
      }

      return 0
    })
    .reduce((sortedDictionary: IDictionary, definition) => {
      sortedDictionary[definition[0]] = definition[1]

      return sortedDictionary
    }, {})

const theme: ITheme = {
  breakpoints: sort(breakpoints),
  colors,
  spacers: sort(spacers),
}

// Exports
export default theme
