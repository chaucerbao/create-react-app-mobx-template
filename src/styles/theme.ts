// Styles
export default {
  black: '#000',
  blue: 'steelblue',
  darkblue: 'royalblue',
  darkgray: '#333',
  dimgray: 'dimgray',
  lightblue: 'lightsteelblue',
  lightgray: 'gainsboro',
  red: 'red',
  white: '#fff'
}
