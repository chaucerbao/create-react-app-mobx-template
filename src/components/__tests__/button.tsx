// Third-party dependencies
import * as React from 'react'
import * as renderer from 'react-test-renderer'
import { ThemeProvider } from 'styled-components'

// Local dependencies
import theme from '../../styles/theme'

// Component
import Button, { LinkButton } from '../button'

// Mocks
const stores = {}

describe('<Button />', () => {
  it('renders a neutral Button', () => {
    const component = renderer
      .create(
        <ThemeProvider theme={theme}>
          <Button>Button Text</Button>
        </ThemeProvider>
      )
      .toJSON()

    expect(component).toMatchSnapshot()
  })

  it('renders a disabled, neutral Button', () => {
    const component = renderer
      .create(
        <ThemeProvider theme={theme}>
          <Button disabled={true}>Button Text</Button>
        </ThemeProvider>
      )
      .toJSON()

    expect(component).toMatchSnapshot()
  })

  it('renders a primary Button', () => {
    const component = renderer
      .create(
        <ThemeProvider theme={theme}>
          <Button primary={true}>Button Text</Button>
        </ThemeProvider>
      )
      .toJSON()

    expect(component).toMatchSnapshot()
  })

  it('renders a disabled, primary Button', () => {
    const component = renderer
      .create(
        <ThemeProvider theme={theme}>
          <Button primary={true} disabled={true}>
            Button Text
          </Button>
        </ThemeProvider>
      )
      .toJSON()

    expect(component).toMatchSnapshot()
  })
})

describe('<LinkButton />', () => {
  it('renders a neutral LinkButton', () => {
    const component = renderer
      .create(
        <ThemeProvider theme={theme}>
          <LinkButton to="/url" stores={stores}>
            Button Text
          </LinkButton>
        </ThemeProvider>
      )
      .toJSON()

    expect(component).toMatchSnapshot()
  })

  it('renders a disabled, neutral LinkButton', () => {
    const component = renderer
      .create(
        <ThemeProvider theme={theme}>
          <LinkButton to="/url" disabled={true} stores={stores}>
            Button Text
          </LinkButton>
        </ThemeProvider>
      )
      .toJSON()

    expect(component).toMatchSnapshot()
  })

  it('renders a primary LinkButton', () => {
    const component = renderer
      .create(
        <ThemeProvider theme={theme}>
          <LinkButton to="/url" primary={true} stores={stores}>
            Button Text
          </LinkButton>
        </ThemeProvider>
      )
      .toJSON()

    expect(component).toMatchSnapshot()
  })

  it('renders a disabled, primary LinkButton', () => {
    const component = renderer
      .create(
        <ThemeProvider theme={theme}>
          <LinkButton to="/url" primary={true} disabled={true} stores={stores}>
            Button Text
          </LinkButton>
        </ThemeProvider>
      )
      .toJSON()

    expect(component).toMatchSnapshot()
  })
})
