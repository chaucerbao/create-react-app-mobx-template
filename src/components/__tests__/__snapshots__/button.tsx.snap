// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> renders a disabled, neutral Button 1`] = `
.c0 {
  -webkit-transition: background 0.2s;
  transition: background 0.2s;
  margin: 0;
  outline: none;
  border-radius: 0;
  padding: 4px 8px;
  font: inherit;
  border: 1px solid #e9e9e9;
  background: #e9e9e9;
  cursor: not-allowed;
  color: #fff;
}

.c0:hover {
  border-color: #e9e9e9;
  background: #e9e9e9;
}

.c0:focus {
  border-color: #e9e9e9;
}

.c0:active {
  background: #e9e9e9;
}

<button
  className="c0"
  disabled={true}
  primary={undefined}
>
  Button Text
</button>
`;

exports[`<Button /> renders a disabled, primary Button 1`] = `
.c0 {
  -webkit-transition: background 0.2s;
  transition: background 0.2s;
  margin: 0;
  outline: none;
  border-radius: 0;
  padding: 4px 8px;
  font: inherit;
  border: 1px solid #8db3d3;
  background: #8db3d3;
  cursor: not-allowed;
  color: #fff;
}

.c0:hover {
  border-color: #8db3d3;
  background: #8db3d3;
}

.c0:focus {
  border-color: #8db3d3;
}

.c0:active {
  background: #8db3d3;
}

<button
  className="c0"
  disabled={true}
  primary={undefined}
>
  Button Text
</button>
`;

exports[`<Button /> renders a neutral Button 1`] = `
.c0 {
  -webkit-transition: background 0.2s;
  transition: background 0.2s;
  margin: 0;
  outline: none;
  border-radius: 0;
  padding: 4px 8px;
  font: inherit;
  border: 1px solid gainsboro;
  background: gainsboro;
  cursor: pointer;
  color: dimgray;
}

.c0:hover {
  border-color: #e9e9e9;
  background: #e9e9e9;
}

.c0:focus {
  border-color: #9c9c9c;
}

.c0:active {
  background: #cfcfcf;
}

<button
  className="c0"
  primary={undefined}
>
  Button Text
</button>
`;

exports[`<Button /> renders a primary Button 1`] = `
.c0 {
  -webkit-transition: background 0.2s;
  transition: background 0.2s;
  margin: 0;
  outline: none;
  border-radius: 0;
  padding: 4px 8px;
  font: inherit;
  border: 1px solid steelblue;
  background: steelblue;
  cursor: pointer;
  color: #fff;
}

.c0:hover {
  border-color: #568ebd;
  background: #568ebd;
}

.c0:focus {
  border-color: #224058;
}

.c0:active {
  background: #3f75a2;
}

<button
  className="c0"
  primary={undefined}
>
  Button Text
</button>
`;

exports[`<LinkButton /> renders a disabled, neutral LinkButton 1`] = `
.c0 {
  -webkit-transition: background 0.2s;
  transition: background 0.2s;
  margin: 0;
  outline: none;
  border-radius: 0;
  padding: 4px 8px;
  font: inherit;
  border: 1px solid #e9e9e9;
  background: #e9e9e9;
  cursor: not-allowed;
  color: #fff;
  text-decoration: none;
}

.c0:hover {
  border-color: #e9e9e9;
  background: #e9e9e9;
}

.c0:focus {
  border-color: #e9e9e9;
}

.c0:active {
  background: #e9e9e9;
}

<a
  className="c0"
  href="/url"
  onClick={[Function]}
  rel={undefined}
>
  Button Text
</a>
`;

exports[`<LinkButton /> renders a disabled, primary LinkButton 1`] = `
.c0 {
  -webkit-transition: background 0.2s;
  transition: background 0.2s;
  margin: 0;
  outline: none;
  border-radius: 0;
  padding: 4px 8px;
  font: inherit;
  border: 1px solid #8db3d3;
  background: #8db3d3;
  cursor: not-allowed;
  color: #fff;
  text-decoration: none;
}

.c0:hover {
  border-color: #8db3d3;
  background: #8db3d3;
}

.c0:focus {
  border-color: #8db3d3;
}

.c0:active {
  background: #8db3d3;
}

<a
  className="c0"
  href="/url"
  onClick={[Function]}
  primary={true}
  rel={undefined}
>
  Button Text
</a>
`;

exports[`<LinkButton /> renders a neutral LinkButton 1`] = `
.c0 {
  -webkit-transition: background 0.2s;
  transition: background 0.2s;
  margin: 0;
  outline: none;
  border-radius: 0;
  padding: 4px 8px;
  font: inherit;
  border: 1px solid gainsboro;
  background: gainsboro;
  cursor: pointer;
  color: dimgray;
  text-decoration: none;
}

.c0:hover {
  border-color: #e9e9e9;
  background: #e9e9e9;
}

.c0:focus {
  border-color: #9c9c9c;
}

.c0:active {
  background: #cfcfcf;
}

<a
  className="c0"
  href="/url"
  onClick={[Function]}
  rel={undefined}
>
  Button Text
</a>
`;

exports[`<LinkButton /> renders a primary LinkButton 1`] = `
.c0 {
  -webkit-transition: background 0.2s;
  transition: background 0.2s;
  margin: 0;
  outline: none;
  border-radius: 0;
  padding: 4px 8px;
  font: inherit;
  border: 1px solid steelblue;
  background: steelblue;
  cursor: pointer;
  color: #fff;
  text-decoration: none;
}

.c0:hover {
  border-color: #568ebd;
  background: #568ebd;
}

.c0:focus {
  border-color: #224058;
}

.c0:active {
  background: #3f75a2;
}

<a
  className="c0"
  href="/url"
  onClick={[Function]}
  primary={true}
  rel={undefined}
>
  Button Text
</a>
`;
